### PR TITLE
Update ember cli to 3.7

### PR DESCRIPTION
**Pull request by patocallaghan**
Using `ember-cli-update` if upgraded this addon to ember-cli 3.7. Most changes are straight-out-of-the-box ember-cli changes, i.e. remove bower, jshint etc The main reason I'm upgrading is to remove the usage of ember-cli-babel 5.x

Things which I needed to do:
* Add yarn.lock
* Migrated to new module API
* Upgraded test to new test syntax
* Replaced usage of `mapProperty` with `getEach`

Brucey Bonus: I've re-activated Travis CI and builds are 🍏 again with this PR

### CODE REVIEW

File: tests/dummy/app/templates/application.hbs [hunk 1]
```diff
@@ -1,10 +1,10 @@
-<h2 id='title'>ember-undo-stack</h2>
+<h2 id="title">ember-undo-stack</h2>
```
Comment (patocallaghan) [inline tests/dummy/app/templates/application.hbs]: Fixes template linting errors. Use `"` instead of `'`

File: tests/unit/undo-stack-test.js [hunk 2]
```diff
@@ -1,230 +1,230 @@
...
-      kittens.pushObject(Cat.restoreCheckpoint(kittenData));
-    });
-    return this;
-  }
-}).reopenClass({
-  deserialize(data) {
-    return Cat.create().deserialize(data);
-  }
-});
+module('UndoStack', function() {
```
Comment (patocallaghan) [inline tests/unit/undo-stack-test.js]: * Uses new module API
* Migrates to new test syntax

File: tests/unit/undo-stack-test.js [hunk 3]
```diff
@@ -1,230 +1,230 @@
...
+      return `${name} (${age}) has ${kittensLength} kittens`;
+    }),
+    setup: on('init', function() {
+      this.set('kittens', A());
+    }),
+    checkpointData: computed('name', 'age', 'kittens.@.checkpointData', function() {
+      return {
+        name: this.get('name'),
+        age: this.get('age'),
+        kittens: this.get('kittens').getEach('checkpointData')
```
Comment (patocallaghan) [inline tests/unit/undo-stack-test.js]: Replace `mapProperty` with `getEach` as it no longer seems to be a thing

File: addon/undo-stack.js [hunk 4]
```diff
@@ -1,23 +1,26 @@
-import Em from 'ember';
-
-const { computed, on, run } = Em;
-
-export default Em.Mixin.create({
+import { notEmpty, not, gt } from '@ember/object/computed';
```
Comment (patocallaghan) [inline addon/undo-stack.js]: Migrates to use the new module API

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+module.exports = {
```
Comment (sduquej) [inline .eslintrc.js]: just a thought while reviewing this, have we considered shared `eslint`/`templatelint`/`prettier`/etc definitions across repos?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# How To Contribute
```
Comment (sduquej) [inline CONTRIBUTING.md]: :zap:?

File: tests/dummy/app/templates/application.hbs (hunk 1 above)
Comment (sduquej) [reply tests/dummy/app/templates/application.hbs]: was this manual or is there a codemod? IIRC @pizza 	tried to do this once in the embercom codebase with regexes and got into rebase hell

File: .eslintrc.js (hunk 5 above)
Comment (patocallaghan) [reply .eslintrc.js]: Yes I've thought about how best to achieve that. Could potentially put all the config in https://github.com/intercom/eslint-plugin-intercom or something else like a new eslint-plugin-intercom-recommended?

**Pull request merged**